From ace49900d00722305e84f505122b8d7812a567ad Mon Sep 17 00:00:00 2001
From: Jianhui Zhao <jianhui.zhao@gl-inet.com>
Date: Thu, 29 Apr 2021 10:17:15 +0800
Subject: [PATCH] feeds: add gl-oui

Signed-off-by: Jianhui Zhao <jianhui.zhao@gl-inet.com>
---
 feeds.conf.default | 1 +
 1 file changed, 1 insertion(+)

diff --git a/feeds.conf.default b/feeds.conf.default
index ae0ba003fc..b9db257475 100644
--- a/feeds.conf.default
+++ b/feeds.conf.default
@@ -1,4 +1,5 @@
 src-git gl https://github.com/gl-inet/gl-feeds.git;19.07.7
+src-git oui https://gitlab.com/gl.sdk4.0/gl.common/gl-oui.git
 src-git packages https://git.openwrt.org/feed/packages.git^90af10d01579425369bd474051b6d3ddaf32a7e3
 src-git luci https://git.openwrt.org/project/luci.git^34e0d656a41befd9720be35c4831c9f136a67c59
 src-git routing https://git.openwrt.org/feed/routing.git^e26b4745209655976b7d124465b1dc53ade632f9
-- 
2.17.1

