From dc9e49fcaa0b9660dc81611748384b75d69759b5 Mon Sep 17 00:00:00 2001
From: Yu Zhang <yu.zhang@gl-inet.com>
Date: Fri, 11 Aug 2023 16:44:34 +0800
Subject: [PATCH] adapt_gl-sip

Signed-off-by: Yu Zhang <yu.zhang@gl-inet.com>
---
 include/netfilter.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/netfilter.mk b/include/netfilter.mk
index 60f031e9a7..e421bc0541 100644
--- a/include/netfilter.mk
+++ b/include/netfilter.mk
@@ -215,8 +215,6 @@ $(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_CONNTRACK_H323, $(P_XT)nf_conn
 $(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_NAT_H323, $(P_V4)nf_nat_h323))
 $(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_CONNTRACK_PPTP, $(P_XT)nf_conntrack_pptp))
 $(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_NAT_PPTP, $(P_V4)nf_nat_pptp))
-$(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_CONNTRACK_SIP, $(P_XT)nf_conntrack_sip))
-$(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_NAT_SIP, $(P_XT)nf_nat_sip))
 $(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_CONNTRACK_SNMP, $(P_XT)nf_conntrack_snmp))
 $(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_NAT_SNMP_BASIC, $(P_V4)nf_nat_snmp_basic))
 $(eval $(call nf_add,NF_NATHELPER_EXTRA,CONFIG_NF_CONNTRACK_TFTP, $(P_XT)nf_conntrack_tftp))
-- 
2.25.1

