From b82329e7306bb158aea9be9ffbdbe1de84e1ecf9 Mon Sep 17 00:00:00 2001
From: niyin <yin.ni@gl-inet.com>
Date: Thu, 8 Dec 2022 19:06:15 -0800
Subject: [PATCH] update LINUX_VERMAGIC

---
 include/kernel.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/kernel.mk b/include/kernel.mk
index 070d1bca75..5f3b4afc2d 100644
--- a/include/kernel.mk
+++ b/include/kernel.mk
@@ -54,6 +54,7 @@ else
     LINUX_UNAME_VERSION:=$(LINUX_UNAME_VERSION)-$(strip $(lastword $(subst -, ,$(LINUX_VERSION))))
   endif
 
+  LINUX_VERMAGIC:=86cd10224392f0f1f47de97edd009387
   LINUX_KERNEL:=$(KERNEL_BUILD_DIR)/vmlinux
 
   ifneq (,$(findstring -rc,$(LINUX_VERSION)))
-- 
2.25.1

