From 942c70abe941c628b2334155c77a662a7ef9a2c2 Mon Sep 17 00:00:00 2001
From: Weiping Yang <weiping.yang@gl-inet.com>
Date: Wed, 8 Mar 2023 03:35:00 -0500
Subject: [PATCH] update 21.02 feeds.conf.default feeds

---
 feeds.conf.default | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/feeds.conf.default b/feeds.conf.default
index e8d9548f60..1db3438e2f 100644
--- a/feeds.conf.default
+++ b/feeds.conf.default
@@ -1,4 +1,4 @@
-src-git packages https://git.openwrt.org/feed/packages.git;openwrt-21.02
-src-git luci https://git.openwrt.org/project/luci.git;openwrt-21.02
-src-git routing https://git.openwrt.org/feed/routing.git;openwrt-21.02
-src-git telephony https://git.openwrt.org/feed/telephony.git;openwrt-21.02
+src-git packages https://git.openwrt.org/feed/packages.git^477e23e85dfac98ba2372fecdcbf2013584a1542
+src-git luci https://git.openwrt.org/project/luci.git^a0c73faf502684bca2ffbdab511510f37d799b76
+src-git routing https://git.openwrt.org/feed/routing.git^83bb4b5d97a3af49d561d469be965c427bfc0db2
+src-git telephony https://git.openwrt.org/feed/telephony.git^920fbc5c0a2e4badf51bceff42e9a1e3eb693462
-- 
2.17.1

