Index: dnsmasq-2.85/src/rfc2131.c
===================================================================
--- dnsmasq-2.85.orig/src/rfc2131.c
+++ dnsmasq-2.85/src/rfc2131.c
@@ -724,7 +724,7 @@ size_t dhcp_reply(struct dhcp_context *c
 	borken_opt = 1;
       else
 	daemon->dhcp_buff[len] = 0;
-      if (legal_hostname(daemon->dhcp_buff))
+      // if (legal_hostname(daemon->dhcp_buff))
 	client_hostname = daemon->dhcp_buff;
     }
 
