From 2ddfb93dbef480f82786ec271d8a7699ff35bc96 Mon Sep 17 00:00:00 2001
From: gl-luochongjun <luochongjun@gl-inet.com>
Date: Sat, 13 Aug 2022 17:03:29 +0800
Subject: [PATCH] fix overlay fs node  invalid

---
 include/image.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/image.mk b/include/image.mk
index 92d343c6b7..d7346ae5dc 100644
--- a/include/image.mk
+++ b/include/image.mk
@@ -226,7 +226,7 @@ $(eval $(foreach S,$(NAND_BLOCKSIZE),$(call Image/mkfs/jffs2-nand/template,$(S))
 
 define Image/mkfs/squashfs-common
 	$(STAGING_DIR_HOST)/bin/mksquashfs4 $(call mkfs_target_dir,$(1)) $@ \
-		-nopad -noappend -root-owned \
+		-nopad -no-exports -noappend -root-owned \
 		-comp $(SQUASHFSCOMP) $(SQUASHFSOPT)
 endef
 
-- 
2.25.1

