From cac92510ab8eaa541003d97dea4b263b79e787f7 Mon Sep 17 00:00:00 2001
From: Luo Chongjun <luochongjun@gl-inet.com>
Date: Thu, 4 May 2023 19:12:26 +0800
Subject: [PATCH] ath79 adjust 02_network

---
 .../nand/base-files/etc/board.d/02_network    | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/target/linux/ath79/nand/base-files/etc/board.d/02_network b/target/linux/ath79/nand/base-files/etc/board.d/02_network
index 3c5627e2a4..054b569b21 100644
--- a/target/linux/ath79/nand/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/nand/base-files/etc/board.d/02_network
@@ -69,6 +69,14 @@ ath79_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0@eth0" "1:lan" "2:lan" "3:lan" "4:lan" "5:wan" "6@eth1"
 		;;
+	glinet,gl-ar300m-nor|\
+	glinet,gl-ar300m-nand|\
+	glinet,gl-ar750|\
+	glinet,gl-x750-nor|\
+	glinet,gl-x300b-nor|\
+	glinet,gl-x300b-nor-nand)
+		ucidef_set_interfaces_lan_wan "eth1" "eth0"
+		;;
 	*)
 		ucidef_set_interfaces_lan_wan "eth0" "eth1"
 		;;
@@ -92,6 +100,17 @@ ath79_setup_macs()
 	netgear,wndr4500-v3)
 		wan_mac=$(mtd_get_mac_binary caldata 0x6)
 		;;
+	glinet,gl-x300b-nor|\
+	glinet,gl-x300b-nor-nand)
+		wan_mac=$(mtd_get_mac_binary art 0x0)
+		lan_mac=$(macaddr_add "$wan_mac" 1)
+		;;
+	glinet,gl-x750-nor|\
+	glinet,gl-ar750s-nor|\
+	glinet,gl-ar750s-nor-nand)
+		wan_mac=$(mtd_get_mac_binary art 0x0)
+		lan_mac=$(macaddr_add "$wan_mac" 1)
+		;;
 	esac
 
 	[ -n "$lan_mac" ] && ucidef_set_interface_macaddr "lan" $lan_mac
-- 
2.25.1

