From 82d31e0c854771dbef5b28d9b225bc81599b7d75 Mon Sep 17 00:00:00 2001
From: Jianhui Zhao <jianhui.zhao@gl-inet.com>
Date: Sat, 28 May 2022 17:33:42 +0800
Subject: [PATCH] wifi-ax: update bdf for gl-axt1800

Signed-off-by: Jianhui Zhao <jianhui.zhao@gl-inet.com>
---
 feeds/wifi-ax/ath11k-wifi/Makefile | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/feeds/wifi-ax/ath11k-wifi/Makefile b/feeds/wifi-ax/ath11k-wifi/Makefile
index 55f46672..830e34e7 100644
--- a/feeds/wifi-ax/ath11k-wifi/Makefile
+++ b/feeds/wifi-ax/ath11k-wifi/Makefile
@@ -36,6 +36,7 @@ ALLWIFIBOARDS:= \
 	wallys-dr6018-v4 \
 	tplink-ex227 \
 	tplink-ex447 \
+	gl-ax1800 \
 	gl-axt1800
 
 ALLWIFIPACKAGES:=$(foreach BOARD,$(ALLWIFIBOARDS),ath11k-wifi-$(BOARD))
@@ -73,6 +74,11 @@ $(call Package/ath11k-wifi-default)
     TITLE:=cig-wf196 6G bdf
 endef
 
+define Package/ath11k-wifi-gl-ax1800
+$(call Package/ath11k-wifi-default)
+    TITLE:=gl-ax1800 bdf
+endef
+
 define Package/ath11k-wifi-gl-axt1800
 $(call Package/ath11k-wifi-default)
     TITLE:=gl-axt1800 bdf
@@ -146,6 +152,11 @@ define Package/ath11k-wifi-cig-wf196_6g/install
 	$(INSTALL_DATA) ./board-cig-wf196_6g.bin.QCN9074 $(1)/lib/firmware/ath11k/QCN9074/hw1.0/board.bin
 endef
 
+define Package/ath11k-wifi-gl-ax1800/install
+	$(INSTALL_DIR) $(1)/lib/firmware/ath11k/IPQ6018/hw1.0/
+	$(INSTALL_DATA) ./board-gl-ax1800.bin.IPQ6018 $(1)/lib/firmware/ath11k/IPQ6018/hw1.0/board-2.bin
+endef
+
 define Package/ath11k-wifi-gl-axt1800/install
 	$(INSTALL_DIR) $(1)/lib/firmware/ath11k/IPQ6018/hw1.0/
 	$(INSTALL_DATA) ./board-gl-axt1800.bin.IPQ6018 $(1)/lib/firmware/ath11k/IPQ6018/hw1.0/board-2.bin
-- 
2.25.1

