Index: ubus-2021-08-09-a72457b6/lua/ubus.c
===================================================================
--- ubus-2021-08-09-a72457b6.orig/lua/ubus.c
+++ ubus-2021-08-09-a72457b6/lua/ubus.c
@@ -393,12 +393,10 @@ static int ubus_lua_load_methods(lua_Sta
 	int pidx = 0;
 
 	/* get the function pointer */
-	lua_pushinteger(L, 1);
-	lua_gettable(L, -2);
+	lua_rawgeti(L, -1, 1);
 
 	/* get the policy table */
-	lua_pushinteger(L, 2);
-	lua_gettable(L, -3);
+	lua_rawgeti(L, -2, 2);
 
 	/* check if the method table is valid */
 	if ((lua_type(L, -2) != LUA_TFUNCTION) ||
