From 900aa5a2fee21b900345e8f6b8469d87ee8fd3d7 Mon Sep 17 00:00:00 2001
From: Jianhui Zhao <jianhui.zhao@gl-inet.com>
Date: Fri, 21 Jan 2022 18:13:29 +0800
Subject: [PATCH 1/4] feed: add gl

Signed-off-by: Jianhui Zhao <jianhui.zhao@gl-inet.com>
---
 feeds.conf.default | 1 +
 1 file changed, 1 insertion(+)

diff --git a/feeds.conf.default b/feeds.conf.default
index 3d24cb7c27..ba66de39c3 100644
--- a/feeds.conf.default
+++ b/feeds.conf.default
@@ -1,3 +1,4 @@
+src-git gl https://github.com/gl-inet/gl-feeds.git;21.02
 src-git packages https://git.openwrt.org/feed/packages.git;openwrt-21.02
 src-git routing https://git.openwrt.org/feed/routing.git;openwrt-21.02
 src-git telephony https://git.openwrt.org/feed/telephony.git;openwrt-21.02
-- 
2.25.1

