From 6fc90747b12f61a38ba4e7ef333b2f104b45ee32 Mon Sep 17 00:00:00 2001
From: Jianhui Zhao <jianhui.zhao@gl-inet.com>
Date: Fri, 21 Jan 2022 18:14:15 +0800
Subject: [PATCH 2/4] feed: add luci

Signed-off-by: Jianhui Zhao <jianhui.zhao@gl-inet.com>
---
 feeds.conf.default | 1 +
 1 file changed, 1 insertion(+)

diff --git a/feeds.conf.default b/feeds.conf.default
index ba66de39c3..5958942e8d 100644
--- a/feeds.conf.default
+++ b/feeds.conf.default
@@ -2,4 +2,5 @@ src-git gl https://github.com/gl-inet/gl-feeds.git;21.02
 src-git packages https://git.openwrt.org/feed/packages.git;openwrt-21.02
 src-git routing https://git.openwrt.org/feed/routing.git;openwrt-21.02
 src-git telephony https://git.openwrt.org/feed/telephony.git;openwrt-21.02
+src-git luci https://git.openwrt.org/project/luci.git;openwrt-21.02
 #src-link custom /usr/src/openwrt/custom-feed
-- 
2.25.1

