From f0cb3a04d08608cf17f26e9557292df740daa95e Mon Sep 17 00:00:00 2001
From: "GL.iNet-Xinfa.Deng" <xinfa.deng@gl-inet.com>
Date: Sat, 12 Feb 2022 14:41:56 +0800
Subject: [PATCH] target: change name nand-factory.img

---
 target/linux/ipq807x/image/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/target/linux/ipq807x/image/Makefile b/target/linux/ipq807x/image/Makefile
index 3a219e6..49c6cc5 100644
--- a/target/linux/ipq807x/image/Makefile
+++ b/target/linux/ipq807x/image/Makefile
@@ -13,9 +13,9 @@ define Device/Default
   ROOTFSNAME_IN_UBI := ubi_rootfs
   BLOCKSIZE := 128k
   PAGESIZE := 2048
-  IMAGES := sysupgrade.tar nand-factory.bin
+  IMAGES := sysupgrade.tar nand-factory.img
   IMAGE/sysupgrade.tar := sysupgrade-tar | append-metadata
-  IMAGE/nand-factory.bin := append-ubi | qsdk-ipq-factory-nand
+  IMAGE/nand-factory.img := append-ubi | qsdk-ipq-factory-nand
   KERNEL_NAME := Image
   KERNEL = kernel-bin | gzip | fit gzip $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb
   KERNEL_INITRAMFS = kernel-bin | gzip | fit gzip $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb
-- 
2.7.4

