From aaac471b77a9befd676ed849a096b85c024b6298 Mon Sep 17 00:00:00 2001
From: Jianhui Zhao <jianhui.zhao@gl-inet.com>
Date: Wed, 11 May 2022 12:05:37 +0800
Subject: [PATCH] lua: use INT64

Signed-off-by: Jianhui Zhao <jianhui.zhao@gl-inet.com>
---
 package/utils/lua/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/utils/lua/Makefile b/package/utils/lua/Makefile
index e15142d8d2..e36bf915be 100644
--- a/package/utils/lua/Makefile
+++ b/package/utils/lua/Makefile
@@ -90,7 +90,7 @@ endef
 define Build/Configure
 endef
 
-TARGET_CFLAGS += -DLUA_USE_LINUX $(FPIC) -std=gnu99
+TARGET_CFLAGS += -DLUA_USE_LINUX $(FPIC) -std=gnu99 -DLNUM_INT64
 
 define Build/Compile
 	$(MAKE) $(PKG_JOBS) -C $(PKG_BUILD_DIR) \
-- 
2.25.1

